widgets/message-subscriber/src/actions/query-action.ts

```diff
@@ -1,31 +1,37 @@
-import {AbstractMessageAction, MessageType, Message, getAppStore, appActions, StringSelectionChangeMessage, DataRecordsSelectionChangeMessage} from 'jimu-core';
+import { AbstractMessageAction, MessageType, Message, MessageDescription, getAppStore, appActions, StringSelectionChangeMessage, DataRecordsSelectionChangeMessage } from 'jimu-core'
 
-export default class QueryAction extends AbstractMessageAction{
-  filterMessageType(messageType: MessageType, messageWidgetId?: string): boolean{
-    return [MessageType.StringSelectionChange, MessageType.DataRecordsSelectionChange].indexOf(messageType) > -1;
+export default class QueryAction extends AbstractMessageAction {
+  filterMessageType (messageType: MessageType, messageWidgetId?: string): boolean {
+    return [MessageType.StringSelectionChange, MessageType.DataRecordsSelectionChange].includes(messageType)
   }
 
-  filterMessage(message: Message): boolean{return true; }
+  // Needed to indicate whether or not the type of message can trigger the filter
+  // message action. Used in the builder to filter the available actions.
+  filterMessageDescription (messageDescription: MessageDescription): boolean {
+    return messageDescription.messageType === MessageType.DataRecordsSelectionChange
+  }
+
+  filterMessage (message: Message): boolean { return true }
 
   //set action setting uri
-  getSettingComponentUri(messageType: MessageType, messageWidgetId?: string): string {
-    return 'actions/query-action-setting';
+  getSettingComponentUri (messageType: MessageType, messageWidgetId?: string): string {
+    return 'actions/query-action-setting'
   }
 
-  onExecute(message: Message, actionConfig?: any): Promise<boolean> | boolean{
+  onExecute (message: Message, actionConfig?: any): Promise<boolean> | boolean {
     let q = `${actionConfig.fieldName} = '${message}'`
-    switch(message.type){
+    switch (message.type) {
```

Comment:
As you remove the `StringSelectionChange` from the `filterMessageDescription`, I think you can remove this switch as here as well. The `StringSelectionChange` is not documented and is not used anywhere now.

Reply:
Thanks, will remove `StringSelectionChange` in upcoming commit.

Reply:
Commit https://github.com/Esri/arcgis-experience-builder-sdk-resources/pull/121/commits/1b2ba8c0eacfe21d7ced22aa22451a69a508505a resolves this.